.github/workflows/bump-version-release.yaml

```diff
@@ -95,3 +95,19 @@ jobs:
           github-token: ${{ secrets.github-token }}
           release-title: "${{ steps.bump-version.outputs.release-version }}"
           release-body: "${{ steps.build-changelog.outputs.single-changelog }}"
+
+      - name: Bump version
+        id: bump-version-snapshot
+        uses: bakdata/ci-templates/actions/bump-version-snapshot@feat/new-bumpversion
+        with:
+          working-directory: ${{ inputs.working-directory }}
+          test-version: ${{ steps.bump-version.outputs.release-version }}
+
+      - name: Commit and push changes including .bumpversion.cfg file
+        uses: bakdata/ci-templates/actions/commit-and-push@v1.6.0
+        with:
+          ref: ${{ github.event.repository.default_branch }}
+          commit-message: "Bump version ${{ steps.bump-version.outputs.release-version  }} → ${{ steps.bump-version-snapshot.outputs.release-version }}"
+          github-username: ${{ secrets.github-username }}
+          github-email: ${{ secrets.github-email }}
+          github-token: ${{ secrets.github-token }}
```

Comment:
We currently need 2x commit and push in this workflow, because we are creating the GitHub release in between, which is not ideal. As the [docs of the softprops/action-gh-release](https://github.com/softprops/action-gh-release?tab=readme-ov-file#-limit-releases-to-pushes-to-tags) action (which is part of our tag-and-release action above) mention, it would be more ideal if the GitHub release is actually created in the workflow running on the tag later with an `if: startsWith(github.ref, 'refs/tags/')` condition. This way we could avoid pushing and in turn triggering CI multiple times on the defualt branch.

Reply:
I still think we shouldn't create the GitHub release in the CI running on the default branch